lmp/tokenizer/_base_dict_tokenizer.py

```diff
@@ -277,9 +302,21 @@ def decode(
                 tokens except unknown word's token.
                 See class docstring for more details on special tokens.
 
+        Raises:
+            TypeError:
+                When `token_ids` is not instance of `List[int]`.
+                When `remove_special_tokens` is not instance of `bool`.
+
         Returns:
             Sequence decoded from `token_ids`.
         """
+        # Type check.
+        if not isinstance(token_ids, List):
+            raise TypeError('`token_ids` must be instance of `List[int]`.')
```

Comment:
First do `token_ids = list(token_ids)`, then use `for` loop to check each element in `token_ids` is instance of `int`.